tests/unit/test_charm.py

```diff
@@ -101,24 +126,36 @@ def test_on_leader_elected(self):
             self.charm._peers.data[self.charm.app].get("postgres-password", None), password
         )
 
+    @patch_network_get(private_address="1.1.1.1")
+    @patch("charm.Patroni.update_cluster_members")
+    @patch("charm.Patroni.member_started")
     @patch("charm.Patroni.bootstrap_cluster")
-    @patch(
-        "charm.PostgresqlOperatorCharm._replication_password", return_value="fake-replication-pw"
-    )
-    @patch("charm.PostgresqlOperatorCharm._get_postgres_password", return_value=None)
-    def test_on_start(self, _get_postgres_password, _replication_password, _bootstrap_cluster):
+    @patch("charm.PostgresqlOperatorCharm._replication_password")
+    @patch("charm.PostgresqlOperatorCharm._get_postgres_password")
+    def test_on_start(
+        self,
+        _get_postgres_password,
+        _replication_password,
+        _bootstrap_cluster,
+        _member_started,
+        _,
+    ):
         # Test before the passwords are generated.
+        _get_postgres_password.return_value = None
         self.charm.on.start.emit()
         _bootstrap_cluster.assert_not_called()
         self.assertTrue(isinstance(self.harness.model.unit.status, WaitingStatus))
 
-        # Mock the generated superuser password.
+        # Mock the passwords.
         _get_postgres_password.return_value = "fake-postgres-password"
+        _replication_password.return_value = "fake-replication-password"
 
         # Mock cluster start success values.
         _bootstrap_cluster.side_effect = [False, True]
 
         # Test for a failed cluster bootstrapping.
+        # TODO: test replicas start.
```

Comment:
Is this for this PR, or will it be resolved in a future one?

Reply:
It is for the future. Added the JIRA issue in https://github.com/canonical/postgresql-operator/pull/11/commits/d2529905daa730a8960c697da611a2e396d8fcf7,